src/python/aps.py

```diff
@@ -0,0 +1,345 @@
+import numpy as np
+import numpy.ctypeslib as npct
+from ctypes import c_int, c_uint, c_ulong, c_ulonglong, c_float, c_double, c_char, c_char_p, addressof, create_string_buffer, byref, POINTER
+
+libaps2 = npct.load_library("libaps2", ".")
+
+np_double_1D = npct.ndpointer(dtype=np.double,  ndim=1, flags='CONTIGUOUS')
+np_float_1D  = npct.ndpointer(dtype=np.float,   ndim=1, flags='CONTIGUOUS')
+np_int16_1D  = npct.ndpointer(dtype=np.int16,   ndim=1, flags='CONTIGUOUS')
+np_int8_1D   = npct.ndpointer(dtype=np.int8,    ndim=1, flags='CONTIGUOUS')
+np_uint64_1D = npct.ndpointer(dtype=np.uint64,  ndim=1, flags='CONTIGUOUS')
+np_uint32_1D = npct.ndpointer(dtype=np.uint32,  ndim=1, flags='CONTIGUOUS')
+np_uint16_1D = npct.ndpointer(dtype=np.uint16,  ndim=1, flags='CONTIGUOUS')
+np_uint8_1D  = npct.ndpointer(dtype=np.uint8,   ndim=1, flags='CONTIGUOUS')
+
+class Flag(object):
+    flags = [(0x1, 'fun'), (0x2, 'toy')]
+    @classmethod
+    def from_param(cls, data):
+        return c_uint(encode_flags(self.flags, data))
+
+# APS2_TRIGGER_SOURCE
```

Comment:
Blake was saying we might look into cffi, though I'm not sure how that matches up yet. Multiple cursors did about 80% of the work, luckily. Swig doesn't actually seem to make things any easier in this case.